Source/Core/Common/Intrinsics.h

```diff
@@ -3,69 +3,86 @@
- * SSE 4.2 on x86/x64.
- */
-
 #if defined(__GNUC__) || defined(__clang__)
+// Due to limitations in GCC, intrinsics are only available when compiling with the corresponding
+// instruction set enabled. However, using the target attribute, we can compile single functions
+// with a different target instruction set, while still creating a generic build.
+//
+// Since this instruction set is enabled per-function, any callers should verify that the
+// instruction set is supported at runtime before calling it, and provide a fallback implementation
+// when not supported.
+//
+// When building with -march=native, or enabling the instruction sets in the compile flags, permit
+// usage of the instrinsics without any function attributes. If the command-line architecture does
+// not support this instruction set, enable it via function targeting.
 
-/**
- * Due to limitations in GCC, SSE intrinsics are only available when compiling with the
- * corresponding instruction set enabled. However, using the target attribute, we can compile
- * single functions with a different target instruction set, while still creating a generic build.
- *
- * Since this instruction set is enabled per-function, any callers should verify that the
- * instruction set is supported at runtime before calling it, and provide a fallback implementation
- * when not supported.
- *
- * When building with -march=native, or enabling the instruction sets in the compile flags, permit
- * usage of the instrinsics without any function attributes. If the command-line architecture does
- * not support this instruction set, enable it via function targeting.
- */
-
+#if defined(_M_X86)
 #include <x86intrin.h>
-#ifndef __SSE4_2__
-#define FUNCTION_TARGET_SSE42 [[gnu::target("sse4.2")]]
+#ifndef __SSE4_2__  // (-msse4.2) Streaming SIMD Extensions 4.2
+#define TARGET_X86_SSE42 [[gnu::target("sse4.2")]]
 #endif
-#ifndef __SSE4_1__
-#define FUNCTION_TARGET_SSR41 [[gnu::target("sse4.1")]]
+#ifndef __SSE4_1__  // (-msse4.1) Streaming SIMD Extensions 4.1
+#define TARGET_X86_SSR41 [[gnu::target("sse4.1")]]
 #endif
-#ifndef __SSSE3__
-#define FUNCTION_TARGET_SSSE3 [[gnu::target("ssse3")]]
+#ifndef __SSSE3__  // (-mssse3) Supplemental Streaming SIMD Extensions 3
+#define TARGET_X86_SSSE3 [[gnu::target("ssse3")]]
 #endif
-#ifndef __SSE3__
-#define FUNCTION_TARGET_SSE3 [[gnu::target("sse3")]]
+#ifndef __SSE3__  // (-msse3) Streaming SIMD Extensions 3
+#define TARGET_X86_SSE3 [[gnu::target("sse3")]]
+#endif
+#ifndef __SHA__  // (-msha) Intel SHA Extensions
+#define TARGET_X86_SHA [[gnu::target("sha")]]
+#endif
+#ifndef __AES__  // (-maes) Advanced Encryption Standard Instruction Set
+#define TARGET_X86_AES [[gnu::target("aes")]]
 #endif
 
-#elif defined(_MSC_VER) || defined(__INTEL_COMPILER)
+#elif defined(_M_ARM_64)
+// This is a hack to get clang to accept the sha1 intrinsics without modifying cmdline flags.
+#define __ARM_FEATURE_SHA2    // __ARM_FEATURE_CRYPTO is deprecated and "SHA2" flag is the lowest
```

Comment:
This stuff isn’t needed anymore

Reply:
Because the files using it are compiled with the -march=arvm8-a+crypto flag in the CMakeLists.txt?

Reply:
yes

Reply:
Hm... That sort of goes counter to what Common/Intrinsics.h is trying to do, but I did encounter the difficulties of enabling ARM CPU extensions while testing things, so I guess I understand.

Reply:
please leave the features being set on the cmdline/cmake

Reply:
I really wanted to keep things consistent.  You can't say I didn't try.  Clang just totally screws up enabling ARM extensions, while GCC does exactly what you expect from it.
https://godbolt.org/z/9Y1cb7Yad
What a shame